### PR TITLE
Kubelet: Allocatable resources check for negative value

### PR DESCRIPTION
**Pull request by ingvagabund**
When setting kube/system-resources for a node, negative quantities can result in node's allocatable being higher then node's capacity. Let's check the quantity and return error if it is negative.

### CODE REVIEW

**Comment by ingvagabund** [comment]
@ncdc, @timstclair

**Comment by ncdc** [comment]
cc @kubernetes/rh-cluster-infra

File: cmd/kubelet/app/server.go [hunk 1]
```diff
@@ -902,6 +902,9 @@ func parseResourceList(m utilconfig.ConfigurationMap) (api.ResourceList, error)
 			if err != nil {
 				return nil, err
 			}
+			if q.Amount.Sign() == -1 {
+				return nil, fmt.Errorf("resource for %q has negative quantity: %v", k, v)
```
Comment (ncdc) [inline cmd/kubelet/app/server.go]: Maybe rephrase to better indicate it's an error (resource must be greater than or equal to zero, resource cannot be negative, something like that)?

Comment (ingvagabund) [inline cmd/kubelet/app/server.go]: ``"resource quantity for %q cannot be negative: %v"`` ?

Comment (ncdc) [inline cmd/kubelet/app/server.go]: Sure

**Comment by ncdc** [comment]
LGTM

**Comment by timstclair** [comment]
LGTM

**Comment by ncdc** [comment]
Any chance we could get a test case for this?

**Comment by ingvagabund** [comment]
unit tests here https://github.com/kubernetes/kubernetes/pull/23627
